drivers/display/display_gc9x01x.c

```diff
@@ -0,0 +1,745 @@
+
+/* Display data struct */
+struct gc9x01x_data {
+	uint8_t bytes_per_pixel;
+	enum display_pixel_format pixel_format;
+	enum display_orientation orientation;
+};
+
+/* Configuration data struct.*/
+struct gc9x01x_config {
+	struct spi_dt_spec spi;
+	struct gpio_dt_spec cmd_data;
+	struct gpio_dt_spec reset;
+	struct pwm_dt_spec backlight;
+	uint8_t pixel_format;
+	uint16_t rotation;
+	uint16_t x_resolution;
+	uint16_t y_resolution;
+	bool inversion;
+	const void *regs;
+	int (*regs_init_fn)(const struct device *dev);
+};
+
+/* Initialization command data struct  */
+struct gc9x01x_default_init_regs {
+	uint8_t cmd;
+	uint8_t len;
+	uint8_t data[GC9X01X_NUM_DEFAULT_INIT_REGS];
+};
+
+/*
+ * Default initialization commands. There are a lot of undocumented commands
+ * within the manufacturer sample code, that are essential for proper operation of
+ * the display controller
+ */
+static const struct gc9x01x_default_init_regs default_init_regs[] = {
+	{
+		.cmd = 0xEBU,
+		.len = 1U,
+		.data = {0x14U},
+	},
+	{
+		.cmd = 0x84U,
+		.len = 1U,
+		.data = {0x40U},
+	},
+	{
+		.cmd = 0x85U,
+		.len = 1U,
+		.data = {0xFFU},
+	},
+	{
+		.cmd = 0x86U,
+		.len = 1U,
+		.data = {0xFFU},
+	},
+	{
+		.cmd = 0x87U,
+		.len = 1U,
+		.data = {0xFFU},
+	},
+	{
+		.cmd = 0x88U,
+		.len = 1U,
+		.data = {0x0AU},
+	},
+	{
+		.cmd = 0x89U,
+		.len = 1U,
+		.data = {0x21U},
+	},
+	{
+		.cmd = 0x8AU,
+		.len = 1U,
+		.data = {0x00U},
+	},
+	{
+		.cmd = 0x8BU,
+		.len = 1U,
+		.data = {0x80U},
+	},
+	{
+		.cmd = 0x8CU,
+		.len = 1U,
+		.data = {0x01U},
+	},
+	{
+		.cmd = 0x8DU,
+		.len = 1U,
+		.data = {0x01U},
+	},
+	{
+		.cmd = 0x8EU,
+		.len = 1U,
+		.data = {0xFFU},
+	},
+	{
+		.cmd = 0x8FU,
+		.len = 1U,
+		.data = {0xFFU},
+	},
+	{
+		.cmd = 0xB6U,
+		.len = 2U,
+		.data = {0x00U, 0x20U},
+	},
+	{
+		.cmd = 0x90U,
+		.len = 4U,
+		.data = {0x08U, 0x08U, 0x08U, 0x08U},
+	},
+	{
+		.cmd = 0xBDU,
+		.len = 1U,
+		.data = {0x06U},
+	},
+	{
+		.cmd = 0xBCU,
+		.len = 1U,
+		.data = {0x00U},
+	},
+	{
+		.cmd = 0xFFU,
+		.len = 3U,
+		.data = {0x60U, 0x01U, 0x04U},
+	},
+	{
+		.cmd = 0xBEU,
+		.len = 1U,
+		.data = {0x11U},
+	},
+	{
+		.cmd = 0xE1U,
+		.len = 2U,
+		.data = {0x10U, 0x0EU},
+	},
+	{
+		.cmd = 0xDFU,
+		.len = 3U,
+		.data = {0x21U, 0x0CU, 0x02U},
+	},
+	{
+		.cmd = 0xEDU,
+		.len = 2U,
+		.data = {0x1BU, 0x0BU},
+	},
+	{
+		.cmd = 0xAEU,
+		.len = 1U,
+		.data = {0x77U},
+	},
+	{
+		.cmd = 0xCDU,
+		.len = 1U,
+		.data = {0x63U},
+	},
+	{
+		.cmd = 0x70U,
+		.len = 9U,
+		.data = {0x07U, 0x07U, 0x04U, 0x0EU, 0x0FU, 0x09U, 0x07U, 0x08U, 0x03U},
+	},
+	{
+		.cmd = 0x62U,
+		.len = 12U,
+		.data = {0x18U, 0x0DU, 0x71U, 0xEDU, 0x70U, 0x70U, 0x18U, 0x0FU, 0x71U, 0xEFU,
+			 0x70U, 0x70U},
+	},
+	{
+		.cmd = 0x63U,
+		.len = 12U,
+		.data = {0x18U, 0x11U, 0x71U, 0xF1U, 0x70U, 0x70U, 0x18U, 0x13U, 0x71U, 0xF3U,
+			 0x70U, 0x70U},
+	},
+	{
+		.cmd = 0x64U,
+		.len = 7U,
+		.data = {0x28U, 0x29U, 0xF1U, 0x01U, 0xF1U, 0x00U, 0x07U},
+	},
+	{
+		.cmd = 0x66U,
```

Comment:
I found out that only 0x66 and 0x67 commands can't be removed and 0x84 mirrors X but it can be compensated by picking parameters for 0x36 which will also require to figure out how to properly set the requested orientation. The other undocumented commands, when removed, don't seem to affect rendering, at least visually.